pandas/lib.pyx

```diff
@@ -1055,6 +1055,20 @@ def string_array_replace_from_nan_rep(ndarray[object, ndim=1] arr, object nan_re
 
 @cython.boundscheck(False)
 @cython.wraparound(False)
+def object_array_decode_bytes(ndarray[object, ndim=1] arr, object encoding):
+    """Decode any instances of bytes to str in arr using the given encoding."""
```

Comment:
I guess you mean Series.str.decode?  I tried that but it does not work if the Series has some bytes and some strs mixed together.  The function I wrote decodes the bytes and leaves the rest intact.  Since the dtype is object either way, I think we need this.